public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx

```diff
@@ -1,72 +1,194 @@
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { cloneDeep, defaultsDeep } from 'lodash';
 import React from 'react';
 
-import { EventBusSrv, TimeRange, toUtc } from '@grafana/data';
-import { setBackendSrv, TemplateSrv } from '@grafana/runtime';
-import { BackendSrv } from 'app/core/services/backend_srv';
-import { ContextSrv } from 'app/core/services/context_srv';
+import { DataSourcePluginMeta } from '@grafana/data';
+import { QueryEditorMode } from 'app/plugins/datasource/prometheus/querybuilder/shared/types';
 
-import { createLokiDatasource } from '../mocks';
-import { LokiQuery } from '../types';
+import { LokiDatasource } from '../datasource';
+import { EXPLAIN_LABEL_FILTER_CONTENT } from '../querybuilder/components/LokiQueryBuilderExplained';
+import { LokiQuery, LokiQueryType } from '../types';
 
-import { LokiQueryEditor } from './LokiQueryEditor';
+import { LokiQueryEditorSelector } from './LokiQueryEditor';
 
-const createMockRequestRange = (from: string, to: string): TimeRange => {
+jest.mock('@grafana/runtime', () => {
   return {
-    from: toUtc(from, 'YYYY-MM-DD'),
-    to: toUtc(to, 'YYYY-MM-DD'),
-    raw: {
-      from: toUtc(from, 'YYYY-MM-DD'),
-      to: toUtc(to, 'YYYY-MM-DD'),
+    ...jest.requireActual('@grafana/runtime'),
+    reportInteraction: jest.fn(),
+  };
+});
+
+jest.mock('app/core/store', () => {
+  return {
+    get() {
+      return undefined;
+    },
+    set() {},
+    getObject(key: string, defaultValue: unknown) {
+      return defaultValue;
     },
   };
+});
+
+const defaultQuery = {
+  refId: 'A',
+  expr: '{label1="foo", label2="bar"}',
 };
 
-const setup = (propOverrides?: object) => {
-  const mockTemplateSrv: TemplateSrv = {
-    getVariables: jest.fn(),
-    replace: jest.fn(),
-    containsTemplate: jest.fn(),
-    updateTimeRange: jest.fn(),
-  };
-  const datasource = createLokiDatasource(mockTemplateSrv);
-  const onRunQuery = jest.fn();
-  const onChange = jest.fn();
+const datasource = new LokiDatasource(
```

Comment:
Why not using createLokiDatasource()?

Reply:
I didn't change tests at all in this case. I just renames `LokiQueryEditorSelector` -> `LokiQueryEditor`, as previous `LokiQueryEditor` wasn't used anymore and I think `LokiQueryEditorSelector` should be named  `LokiQueryEditor` as that is what it is. I can (in the next PR) update the tests, but in this PR, I just wanted to remove unused things and rename and move `LokiQueryEditorSelector` where it belongs.

Reply:
Ohh I see, I see.